Rename TextIndicatorStyle to TextAnimation.

megangardner opened the pull request:
#### 3cecaf29a7bc9af7252286883e5504c387976a61
<pre>
Rename TextIndicatorStyle to TextAnimation.
<a href="https://bugs.webkit.org/show_bug.cgi?id=275517">https://bugs.webkit.org/show_bug.cgi?id=275517</a>
<a href="https://rdar.apple.com/129878612">rdar://129878612</a>

Reviewed by Richard Robinson.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::removeTextAnimationForID):
(WebCore::ChromeClient::cleanUpTextAnimationsForSessionID):
(WebCore::ChromeClient::addSourceTextAnimation):
(WebCore::ChromeClient::addDestinationTextAnimation):
(WebCore::ChromeClient::removeTextIndicatorStyleForID): Deleted.
(WebCore::ChromeClient::cleanUpTextStylesForSessionID): Deleted.
(WebCore::ChromeClient::addSourceTextIndicatorStyle): Deleted.
(WebCore::ChromeClient::addDestinationTextIndicatorStyle): Deleted.
* Source/WebCore/page/unified-text-replacement/UnifiedTextReplacementController.h:
* Source/WebCore/page/unified-text-replacement/UnifiedTextReplacementController.mm:
(WebCore::UnifiedTextReplacementController::textReplacementSessionDidReceiveReplacements):
(WebCore::UnifiedTextReplacementController::textReplacementSessionDidReceiveTextWithReplacementRange):
(WebCore::UnifiedTextReplacementController::didEndTextReplacementSession):
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/webkit/messages.py:
(headers_for_type):
* Source/WebKit/Shared/TextAnimationType.serialization.in: Renamed from Source/WebKit/Shared/TextIndicatorStyle.serialization.in.
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(toWKTextAnimationType):
(-[WKWebView _addTextAnimationTypeForID:withData:]):
(-[WKWebView _removeTextAnimationForID:]):
(-[WKWebView _enableTextIndicatorStylingAfterElementWithID:]):
(-[WKWebView _enableTextIndicatorStylingForElementWithID:]):
(-[WKWebView _disableTextIndicatorStylingWithUUID:]):
(-[WKWebView _enableSourceTextAnimationAfterElementWithID:]):
(-[WKWebView _enableFinalTextAnimationForElementWithID:]):
(-[WKWebView _disableTextAnimationWithUUID:]):
(toWKTextIndicatorStyleType): Deleted.
(-[WKWebView _addTextIndicatorStyleForID:withData:]): Deleted.
(-[WKWebView _removeTextIndicatorStyleForID:]): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
* Source/WebKit/UIProcess/API/mac/WKWebViewMac.mm:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.h:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.mm:
(WebKit::PageClientImplCocoa::addTextAnimationTypeForID):
(WebKit::PageClientImplCocoa::removeTextAnimationForID):
(WebKit::PageClientImplCocoa::addTextIndicatorStyleForID): Deleted.
(WebKit::PageClientImplCocoa::removeTextIndicatorStyleForID): Deleted.
* Source/WebKit/UIProcess/Cocoa/WKTextAnimationType.h: Renamed from Source/WebKit/UIProcess/Cocoa/WKTextIndicatorStyleType.h.
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.h:
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.mm:
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::enableSourceTextAnimationAfterElementWithID):
(WebKit::WebPageProxy::enableTextAnimationTypeForElementWithID):
(WebKit::WebPageProxy::addTextAnimationTypeForID):
(WebKit::WebPageProxy::getTextIndicatorForID):
(WebKit::WebPageProxy::updateUnderlyingTextVisibilityForTextAnimationID):
(WebKit::WebPageProxy::removeTextAnimationForID):
(WebKit::WebPageProxy::enableTextIndicatorStyleAfterElementWithID): Deleted.
(WebKit::WebPageProxy::enableTextIndicatorStyleForElementWithID): Deleted.
(WebKit::WebPageProxy::addTextIndicatorStyleForID): Deleted.
(WebKit::WebPageProxy::updateTextIndicatorStyleVisibilityForID): Deleted.
(WebKit::WebPageProxy::removeTextIndicatorStyleForID): Deleted.
* Source/WebKit/UIProcess/PageClient.h:
* Source/WebKit/UIProcess/TextAnimationType.h: Renamed from Source/WebKit/UIProcess/TextIndicatorStyle.h.
* Source/WebKit/UIProcess/WKSTextAnimationManager.h: Renamed from Source/WebKit/UIProcess/WKSTextStyleManager.h.
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView addTextAnimationTypeForID:withStyleType:]):
(-[WKContentView removeTextAnimationForID:]):
(-[WKContentView updateUnderlyingTextVisibilityForTextAnimationID:visible:completionHandler:]):
(-[WKContentView containingViewForTextAnimationType]):
(-[WKContentView addTextIndicatorStyleForID:withStyleType:]): Deleted.
(-[WKContentView removeTextIndicatorStyleForID:]): Deleted.
(-[WKContentView updateTextIndicatorStyleVisibilityForID:visible:completionHandler:]): Deleted.
(-[WKContentView containingViewForTextIndicatorStyle]): Deleted.
* Source/WebKit/UIProcess/mac/WKTextAnimationManager.h: Renamed from Source/WebKit/UIProcess/mac/WKTextIndicatorStyleManager.h.
* Source/WebKit/UIProcess/mac/WKTextAnimationManager.mm: Renamed from Source/WebKit/UIProcess/mac/WKTextIndicatorStyleManager.mm.
(-[WKTextAnimationTypeEffectData initWithEffectID:type:]):
(-[WKTextAnimationTypeEffectData effectID]):
(-[WKTextAnimationManager initWithWebViewImpl:]):
(-[WKTextAnimationManager addTextAnimationTypeForID:withData:]):
(-[WKTextAnimationManager removeTextAnimationForID:]):
(-[WKTextAnimationManager hasActiveTextAnimationType]):
(-[WKTextAnimationManager suppressTextAnimationType]):
(-[WKTextAnimationManager restoreTextAnimationType]):
(-[WKTextAnimationManager textPreviewsForChunk:completion:]):
(-[WKTextAnimationManager textPreviewForRect:completion:]):
(-[WKTextAnimationManager updateIsTextVisible:forChunk:completion:]):
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::hasContentRelativeChildViews const):
(WebKit::WebViewImpl::suppressContentRelativeChildViews):
(WebKit::WebViewImpl::restoreContentRelativeChildViews):
(WebKit::WebViewImpl::addTextAnimationTypeForID):
(WebKit::WebViewImpl::removeTextAnimationForID):
(WebKit::WebViewImpl::addTextIndicatorStyleForID): Deleted.
(WebKit::WebViewImpl::removeTextIndicatorStyleForID): Deleted.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebKitSwift/TextAnimation/TextAnimationManager.swift: Copied from Source/WebKit/WebKitSwift/TextIndicatorStyle/TextStyleManager.swift.
(currentEffect):
(delegate):
(beginEffect(for:style:)):
(endEffect(for:)):
(TextAnimationManager.targetedPreview(for:)):
(TextAnimationManager.updateTextChunkVisibilityForAnimation(_:visible:)):
(TextAnimationManager.performReplacementAndGeneratePreview(for:effect:animation:)):
(TextAnimationManager.replacementEffectDidComplete(_:)):
* Source/WebKit/WebKitSwift/TextAnimation/TextStyleManager.swift: Renamed from Source/WebKit/WebKitSwift/TextIndicatorStyle/TextStyleManager.swift.
(currentEffect):
(delegate):
(beginEffect(for:style:)):
(endEffect(for:)):
(TextAnimationManager.targetedPreview(for:)):
(TextAnimationManager.updateTextChunkVisibilityForAnimation(_:visible:)):
(TextAnimationManager.performReplacementAndGeneratePreview(for:effect:animation:)):
(TextAnimationManager.replacementEffectDidComplete(_:)):
* Source/WebKit/WebKitSwift/TextAnimation/WKSTextAnimationSourceDelegate.h: Copied from Source/WebKit/WebKitSwift/TextIndicatorStyle/WKSTextStyleSourceDelegate.h.
* Source/WebKit/WebKitSwift/TextAnimation/WKSTextStyleSourceDelegate.h: Renamed from Source/WebKit/WebKitSwift/TextIndicatorStyle/WKSTextStyleSourceDelegate.h.
* Source/WebKit/WebKitSwift/WebKitSwift.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::removeTextAnimationForID):
(WebKit::WebChromeClient::cleanUpTextAnimationsForSessionID):
(WebKit::WebChromeClient::addSourceTextAnimation):
(WebKit::WebChromeClient::addDestinationTextAnimation):
(WebKit::WebChromeClient::removeTextIndicatorStyleForID): Deleted.
(WebKit::WebChromeClient::cleanUpTextStylesForSessionID): Deleted.
(WebKit::WebChromeClient::addSourceTextIndicatorStyle): Deleted.
(WebKit::WebChromeClient::addDestinationTextIndicatorStyle): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebPage/Cocoa/TextAnimationController.mm: Renamed from Source/WebKit/WebProcess/WebPage/Cocoa/TextIndicatorStyleController.mm.
(WebKit::TextAnimationController::TextAnimationController):
(WebKit::TextAnimationController::document const):
(WebKit::TextAnimationController::contextRangeForSessionWithID const):
(WebKit::TextAnimationController::contextRangeForTextAnimationType const):
(WebKit::TextAnimationController::cleanUpTextAnimationsForSessionID):
(WebKit::TextAnimationController::removeTransparentMarkersForTextAnimationID):
(WebKit::newlyReplacedCharacterRange):
(WebKit::TextAnimationController::addSourceTextAnimation):
(WebKit::TextAnimationController::addDestinationTextAnimation):
(WebKit::TextAnimationController::updateUnderlyingTextVisibilityForTextAnimationID):
(WebKit::TextAnimationController::createTextIndicatorForRange):
(WebKit::TextAnimationController::createTextIndicatorForTextAnimationID):
(WebKit::TextAnimationController::enableSourceTextAnimationAfterElementWithID):
(WebKit::TextAnimationController::enableTextAnimationTypeForElementWithID):
* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::createTextIndicatorForTextAnimationID):
(WebKit::WebPage::updateUnderlyingTextVisibilityForTextAnimationID):
(WebKit::WebPage::enableSourceTextAnimationAfterElementWithID):
(WebKit::WebPage::enableTextAnimationTypeForElementWithID):
(WebKit::WebPage::createTextIndicatorForID): Deleted.
(WebKit::WebPage::updateTextIndicatorStyleVisibilityForID): Deleted.
(WebKit::WebPage::enableTextIndicatorStyleAfterElementWithID): Deleted.
(WebKit::WebPage::enableTextIndicatorStyleForElementWithID): Deleted.
* Source/WebKit/WebProcess/WebPage/TextAnimationController.h: Renamed from Source/WebKit/WebProcess/WebPage/TextIndicatorStyleController.h.
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_textAnimationController):
(WebKit::WebPage::addTextAnimationTypeForID):
(WebKit::WebPage::removeTextAnimationForID):
(WebKit::WebPage::cleanUpTextAnimationsForSessionID):
(WebKit::WebPage::addSourceTextAnimation):
(WebKit::WebPage::addDestinationTextAnimation):
(WebKit::m_textIndicatorStyleController): Deleted.
(WebKit::WebPage::addTextIndicatorStyleForID): Deleted.
(WebKit::WebPage::removeTextIndicatorStyleForID): Deleted.
(WebKit::WebPage::cleanUpTextStylesForSessionID): Deleted.
(WebKit::WebPage::addSourceTextIndicatorStyle): Deleted.
(WebKit::WebPage::addDestinationTextIndicatorStyle): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:

Canonical link: <a href="https://commits.webkit.org/280060@main">https://commits.webkit.org/280060@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/709bfd178f6eedece622f2d7fa46c8bc9d75f54f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/55641 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/34964 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/8108 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/58625 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/6071 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/57767 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/42586 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/6270 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/44807 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/4171 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/57670 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/32866 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/47959 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25940 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/55169 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/29652 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/5288 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/4215 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/48718 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/51599 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/5555 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/60216 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/54878 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/30795 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/5680 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/52238 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/31880 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/48029 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/51726 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12320 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/32961 "Built successfully") | | [  ~~🛠 jsc-armv7~~](https://ews-build.webkit.org/#/builders/35/builds/74968 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/31627 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/35/builds/74968 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebCore/page/ChromeClient.h:
```diff
@@ -668,13 +668,13 @@ class ChromeClient {
+    virtual void removeTextAnimationForID(const UnifiedTextReplacement::SessionID&) { }
 
-    virtual void cleanUpTextStylesForSessionID(const UnifiedTextReplacement::SessionID&) { }
+    virtual void cleanUpTextAnimationsForSessionID(const UnifiedTextReplacement::SessionID&) { }
```
I think "clean up" is a bit vague; what does this actually do?

megangardner replied:
It removed all animations associated with this id as well as all visibility markers. It's more of a final... cleanup... than something that should be strictly necessary

rr-codes replied:
is cleanUpTextAnimationsForSessionID ever called without calling removeTextAnimationForID before / after? I'm wodndering if they can just be merged into one method

rr-codes commented on Source/WebCore/page/ChromeClient.h:
```diff
@@ -668,13 +668,13 @@ class ChromeClient {
     virtual void textReplacementSessionUpdateStateForReplacementWithID(const UnifiedTextReplacement::SessionID&, UnifiedTextReplacement::ReplacementState, const UnifiedTextReplacement::ReplacementID&) { }
 
-    virtual void removeTextIndicatorStyleForID(const UnifiedTextReplacement::SessionID&) { }
+    virtual void removeTextAnimationForID(const UnifiedTextReplacement::SessionID&) { }
```
`removeWritingToolsTextAnimationForSessionWithID`?

megangardner replied:
So we went with not using writing Tools for this because I also use them for Safari.

rr-codes replied:
ok, then in that case I think it should be `removeTextAnimationForSessionWithID` (for consistency + clarity)

rr-codes commented on Source/WebCore/page/ChromeClient.h:
```diff
@@ -668,13 +668,13 @@ class ChromeClient {
+    virtual void cleanUpTextAnimationsForSessionID(const UnifiedTextReplacement::SessionID&) { }
 
-    virtual void addSourceTextIndicatorStyle(const UnifiedTextReplacement::SessionID&, const CharacterRange&) { }
+    virtual void addSourceTextAnimation(const UnifiedTextReplacement::SessionID&, const CharacterRange&) { }
```
`addWritingToolsSourceTextAnimationForSessionWithID`? (also, maybe label the CharacterRange parameter, since it's unclear what the parameter is

rr-codes commented on Source/WebCore/page/ChromeClient.h:
```diff
@@ -668,13 +668,13 @@ class ChromeClient {
+    virtual void addSourceTextAnimation(const UnifiedTextReplacement::SessionID&, const CharacterRange&) { }
 
-    virtual void addDestinationTextIndicatorStyle(const UnifiedTextReplacement::SessionID&, const CharacterRange&) { }
+    virtual void addDestinationTextAnimation(const UnifiedTextReplacement::SessionID&, const CharacterRange&) { }
```
`addWritingToolsDestinationTextAnimationForSessionWithID`?

rr-codes commented on Source/WebKit/Shared/TextAnimationType.serialization.in:
```diff
@@ -22,16 +22,16 @@
 
-header: "TextIndicatorStyle.h"
-[CustomHeader] enum class WebKit::TextIndicatorStyle : uint8_t {
+header: "TextAnimationType.h"
```
WritingToolsTextAnimationType?

rr-codes commented on Source/WebKit/Shared/TextAnimationType.serialization.in:
```diff
@@ -22,16 +22,16 @@
-[CustomHeader] enum class WebKit::TextIndicatorStyle : uint8_t {
+header: "TextAnimationType.h"
+[CustomHeader] enum class WebKit::TextAnimationType : uint8_t {
     Initial,
     Source,
```
what's the difference between Initial and Source?

megangardner replied:
Initial is the first "sweep" effect, "source" is the "sweep down" effect. These match the names in Writing Tools

rr-codes commented on Source/WebKit/Shared/TextAnimationType.serialization.in:
```diff
@@ -22,16 +22,16 @@
-[CustomHeader] struct WebKit::TextIndicatorStyleData {
-    WebKit::TextIndicatorStyle style;
+header: "TextAnimationType.h"
+[CustomHeader] struct WebKit::TextAnimationTypeData {
```
maybe just `WritingToolsTextAnimation`?

rr-codes commented on Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
```diff
@@ -2815,6 +2815,19 @@ - (void)_addAppHighlightInNewGroup:(BOOL)newGroup originatedInApp:(BOOL)originat
+{
+    return [self _enableSourceTextAnimationAfterElementWithID:elementID];
+}
+- (NSUUID *)_enableTextIndicatorStylingForElementWithID:(NSString *)elementID
```
is this name (`_enableTextIndicatorStylingForElementWithID`) still accurate?

megangardner replied:
I have these here so that Safari doesn't break, but will not that they all just call the new names :)

rr-codes commented on Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
```diff
@@ -13163,15 +13163,15 @@ - (void)targetedPreviewForID:(NSUUID *)uuid completionHandler:(void (^)(UITarget
 }
 
-- (void)updateTextIndicatorStyleVisibilityForID:(NSUUID *)uuid visible:(BOOL)visible completionHandler:(void (^)(void))completionHandler
+- (void)updateUnderlyingTextVisibilityForTextAnimationID:(NSUUID *)uuid visible:(BOOL)visible completionHandler:(void (^)(void))completionHandler
```
`... forTextAnimationWithID` (for consistency)

rr-codes commented on Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
```diff
@@ -1787,39 +1787,39 @@ - (void)_storeAppHighlight:(const WebCore::AppHighlight&)highlight
 
 #if ENABLE(WRITING_TOOLS_UI)
-- (void)_addTextIndicatorStyleForID:(NSUUID *)nsUUID withData:(const WebKit::TextIndicatorStyleData&)data
+- (void)_addTextAnimationTypeForID:(NSUUID *)nsUUID withData:(const WebKit::TextAnimationTypeData&)data
```
beginEffectForTextAnimationWithID? (to match the UIKit prose)

megangardner committed:
Rename TextIndicatorStyle to TextAnimation.
https://bugs.webkit.org/show_bug.cgi?id=275517
rdar://129878612

Reviewed by Richard Robinson.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::removeTextAnimationForID):
(WebCore::ChromeClient::cleanUpTextAnimationsForSessionID):
(WebCore::ChromeClient::addSourceTextAnimation):
(WebCore::ChromeClient::addDestinationTextAnimation):
(WebCore::ChromeClient::removeTextIndicatorStyleForID): Deleted.
(WebCore::ChromeClient::cleanUpTextStylesForSessionID): Deleted.
(WebCore::ChromeClient::addSourceTextIndicatorStyle): Deleted.
(WebCore::ChromeClient::addDestinationTextIndicatorStyle): Deleted.
* Source/WebCore/page/unified-text-replacement/UnifiedTextReplacementController.h:
* Source/WebCore/page/unified-text-replacement/UnifiedTextReplacementController.mm:
(WebCore::UnifiedTextReplacementController::textReplacementSessionDidReceiveReplacements):
(WebCore::UnifiedTextReplacementController::textReplacementSessionDidReceiveTextWithReplacementRange):
(WebCore::UnifiedTextReplacementController::didEndTextReplacementSession):
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/webkit/messages.py:
(headers_for_type):
* Source/WebKit/Shared/TextAnimationType.serialization.in: Renamed from Source/WebKit/Shared/TextIndicatorStyle.serialization.in.
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(toWKTextAnimationType):
(-[WKWebView _addTextAnimationTypeForID:withData:]):
(-[WKWebView _removeTextAnimationForID:]):
(-[WKWebView _enableTextIndicatorStylingAfterElementWithID:]):
(-[WKWebView _enableTextIndicatorStylingForElementWithID:]):
(-[WKWebView _disableTextIndicatorStylingWithUUID:]):
(-[WKWebView _enableSourceTextAnimationAfterElementWithID:]):
(-[WKWebView _enableFinalTextAnimationForElementWithID:]):
(-[WKWebView _disableTextAnimationWithUUID:]):
(toWKTextIndicatorStyleType): Deleted.
(-[WKWebView _addTextIndicatorStyleForID:withData:]): Deleted.
(-[WKWebView _removeTextIndicatorStyleForID:]): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
* Source/WebKit/UIProcess/API/mac/WKWebViewMac.mm:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.h:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.mm:
(WebKit::PageClientImplCocoa::addTextAnimationTypeForID):
(WebKit::PageClientImplCocoa::removeTextAnimationForID):
(WebKit::PageClientImplCocoa::addTextIndicatorStyleForID): Deleted.
(WebKit::PageClientImplCocoa::removeTextIndicatorStyleForID): Deleted.
* Source/WebKit/UIProcess/Cocoa/WKTextAnimationType.h: Renamed from Source/WebKit/UIProcess/Cocoa/WKTextIndicatorStyleType.h.
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.h:
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.mm:
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::enableSourceTextAnimationAfterElementWithID):
(WebKit::WebPageProxy::enableTextAnimationTypeForElementWithID):
(WebKit::WebPageProxy::addTextAnimationTypeForID):
(WebKit::WebPageProxy::getTextIndicatorForID):
(WebKit::WebPageProxy::updateUnderlyingTextVisibilityForTextAnimationID):
(WebKit::WebPageProxy::removeTextAnimationForID):
(WebKit::WebPageProxy::enableTextIndicatorStyleAfterElementWithID): Deleted.
(WebKit::WebPageProxy::enableTextIndicatorStyleForElementWithID): Deleted.
(WebKit::WebPageProxy::addTextIndicatorStyleForID): Deleted.
(WebKit::WebPageProxy::updateTextIndicatorStyleVisibilityForID): Deleted.
(WebKit::WebPageProxy::removeTextIndicatorStyleForID): Deleted.
* Source/WebKit/UIProcess/PageClient.h:
* Source/WebKit/UIProcess/TextAnimationType.h: Renamed from Source/WebKit/UIProcess/TextIndicatorStyle.h.
* Source/WebKit/UIProcess/WKSTextAnimationManager.h: Renamed from Source/WebKit/UIProcess/WKSTextStyleManager.h.
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView addTextAnimationTypeForID:withStyleType:]):
(-[WKContentView removeTextAnimationForID:]):
(-[WKContentView updateUnderlyingTextVisibilityForTextAnimationID:visible:completionHandler:]):
(-[WKContentView containingViewForTextAnimationType]):
(-[WKContentView addTextIndicatorStyleForID:withStyleType:]): Deleted.
(-[WKContentView removeTextIndicatorStyleForID:]): Deleted.
(-[WKContentView updateTextIndicatorStyleVisibilityForID:visible:completionHandler:]): Deleted.
(-[WKContentView containingViewForTextIndicatorStyle]): Deleted.
* Source/WebKit/UIProcess/mac/WKTextAnimationManager.h: Renamed from Source/WebKit/UIProcess/mac/WKTextIndicatorStyleManager.h.
* Source/WebKit/UIProcess/mac/WKTextAnimationManager.mm: Renamed from Source/WebKit/UIProcess/mac/WKTextIndicatorStyleManager.mm.
(-[WKTextAnimationTypeEffectData initWithEffectID:type:]):
(-[WKTextAnimationTypeEffectData effectID]):
(-[WKTextAnimationManager initWithWebViewImpl:]):
(-[WKTextAnimationManager addTextAnimationTypeForID:withData:]):
(-[WKTextAnimationManager removeTextAnimationForID:]):
(-[WKTextAnimationManager hasActiveTextAnimationType]):
(-[WKTextAnimationManager suppressTextAnimationType]):
(-[WKTextAnimationManager restoreTextAnimationType]):
(-[WKTextAnimationManager textPreviewsForChunk:completion:]):
(-[WKTextAnimationManager textPreviewForRect:completion:]):
(-[WKTextAnimationManager updateIsTextVisible:forChunk:completion:]):
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::hasContentRelativeChildViews const):
(WebKit::WebViewImpl::suppressContentRelativeChildViews):
(WebKit::WebViewImpl::restoreContentRelativeChildViews):
(WebKit::WebViewImpl::addTextAnimationTypeForID):
(WebKit::WebViewImpl::removeTextAnimationForID):
(WebKit::WebViewImpl::addTextIndicatorStyleForID): Deleted.
(WebKit::WebViewImpl::removeTextIndicatorStyleForID): Deleted.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebKitSwift/TextAnimation/TextAnimationManager.swift: Copied from Source/WebKit/WebKitSwift/TextIndicatorStyle/TextStyleManager.swift.
(currentEffect):
(delegate):
(beginEffect(for:style:)):
(endEffect(for:)):
(TextAnimationManager.targetedPreview(for:)):
(TextAnimationManager.updateTextChunkVisibilityForAnimation(_:visible:)):
(TextAnimationManager.performReplacementAndGeneratePreview(for:effect:animation:)):
(TextAnimationManager.replacementEffectDidComplete(_:)):
* Source/WebKit/WebKitSwift/TextAnimation/TextStyleManager.swift: Renamed from Source/WebKit/WebKitSwift/TextIndicatorStyle/TextStyleManager.swift.
(currentEffect):
(delegate):
(beginEffect(for:style:)):
(endEffect(for:)):
(TextAnimationManager.targetedPreview(for:)):
(TextAnimationManager.updateTextChunkVisibilityForAnimation(_:visible:)):
(TextAnimationManager.performReplacementAndGeneratePreview(for:effect:animation:)):
(TextAnimationManager.replacementEffectDidComplete(_:)):
* Source/WebKit/WebKitSwift/TextAnimation/WKSTextAnimationSourceDelegate.h: Copied from Source/WebKit/WebKitSwift/TextIndicatorStyle/WKSTextStyleSourceDelegate.h.
* Source/WebKit/WebKitSwift/TextAnimation/WKSTextStyleSourceDelegate.h: Renamed from Source/WebKit/WebKitSwift/TextIndicatorStyle/WKSTextStyleSourceDelegate.h.
* Source/WebKit/WebKitSwift/WebKitSwift.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::removeTextAnimationForID):
(WebKit::WebChromeClient::cleanUpTextAnimationsForSessionID):
(WebKit::WebChromeClient::addSourceTextAnimation):
(WebKit::WebChromeClient::addDestinationTextAnimation):
(WebKit::WebChromeClient::removeTextIndicatorStyleForID): Deleted.
(WebKit::WebChromeClient::cleanUpTextStylesForSessionID): Deleted.
(WebKit::WebChromeClient::addSourceTextIndicatorStyle): Deleted.
(WebKit::WebChromeClient::addDestinationTextIndicatorStyle): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebPage/Cocoa/TextAnimationController.mm: Renamed from Source/WebKit/WebProcess/WebPage/Cocoa/TextIndicatorStyleController.mm.
(WebKit::TextAnimationController::TextAnimationController):
(WebKit::TextAnimationController::document const):
(WebKit::TextAnimationController::contextRangeForSessionWithID const):
(WebKit::TextAnimationController::contextRangeForTextAnimationType const):
(WebKit::TextAnimationController::cleanUpTextAnimationsForSessionID):
(WebKit::TextAnimationController::removeTransparentMarkersForTextAnimationID):
(WebKit::newlyReplacedCharacterRange):
(WebKit::TextAnimationController::addSourceTextAnimation):
(WebKit::TextAnimationController::addDestinationTextAnimation):
(WebKit::TextAnimationController::updateUnderlyingTextVisibilityForTextAnimationID):
(WebKit::TextAnimationController::createTextIndicatorForRange):
(WebKit::TextAnimationController::createTextIndicatorForTextAnimationID):
(WebKit::TextAnimationController::enableSourceTextAnimationAfterElementWithID):
(WebKit::TextAnimationController::enableTextAnimationTypeForElementWithID):
* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::createTextIndicatorForTextAnimationID):
(WebKit::WebPage::updateUnderlyingTextVisibilityForTextAnimationID):
(WebKit::WebPage::enableSourceTextAnimationAfterElementWithID):
(WebKit::WebPage::enableTextAnimationTypeForElementWithID):
(WebKit::WebPage::createTextIndicatorForID): Deleted.
(WebKit::WebPage::updateTextIndicatorStyleVisibilityForID): Deleted.
(WebKit::WebPage::enableTextIndicatorStyleAfterElementWithID): Deleted.
(WebKit::WebPage::enableTextIndicatorStyleForElementWithID): Deleted.
* Source/WebKit/WebProcess/WebPage/TextAnimationController.h: Renamed from Source/WebKit/WebProcess/WebPage/TextIndicatorStyleController.h.
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_textAnimationController):
(WebKit::WebPage::addTextAnimationTypeForID):
(WebKit::WebPage::removeTextAnimationForID):
(WebKit::WebPage::cleanUpTextAnimationsForSessionID):
(WebKit::WebPage::addSourceTextAnimation):
(WebKit::WebPage::addDestinationTextAnimation):
(WebKit::m_textIndicatorStyleController): Deleted.
(WebKit::WebPage::addTextIndicatorStyleForID): Deleted.
(WebKit::WebPage::removeTextIndicatorStyleForID): Deleted.
(WebKit::WebPage::cleanUpTextStylesForSessionID): Deleted.
(WebKit::WebPage::addSourceTextIndicatorStyle): Deleted.
(WebKit::WebPage::addDestinationTextIndicatorStyle): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:

Canonical link: https://commits.webkit.org/280060@main
